src/imperative/imperative.cc

```diff
@@ -33,11 +33,13 @@ namespace mxnet {
 thread_local bool Imperative::is_train_                 = false;
 thread_local bool Imperative::is_recording_             = false;
 thread_local bool Imperative::is_deferred_compute_      = false;
+thread_local OptConstraint Imperative::opt_constraints_ = OptConstraint::None;
```

Comment:
```suggestion
thread_local unsigned int Imperative::opt_constraints_ = OptConstraint::None;
```
as I understand we are going to store bitmask of the features